src/pythonfinder/models/path.py

```diff
@@ -1,57 +1,34 @@
-from __future__ import annotations
-
 import errno
 import operator
 import os
-import sys
-from collections import defaultdict
-from itertools import chain
 from pathlib import Path
-from typing import (
-    TYPE_CHECKING,
-    Any,
-    Callable,
-    DefaultDict,
-    Generator,
-    Iterator,
-    List,
-    TypeVar,
-    Union,
-)
+import sys
+from collections import defaultdict, ChainMap
+from typing import Any, Dict, List, Generator, Iterator, Optional, Tuple, Union
 
-import attr
 from cached_property import cached_property
+from pydantic import Field, root_validator
 
+from .common import FinderBaseModel
 from ..compat import fs_str
 from ..environment import (
     ASDF_DATA_DIR,
     ASDF_INSTALLED,
     PYENV_INSTALLED,
     PYENV_ROOT,
-    SHIM_PATHS,
     get_shim_paths,
 )
-from ..exceptions import InvalidPythonVersion
 from ..utils import (
     dedup,
```

Comment:
Not an action item here, but I am writing this so we'll remember.
utils.py here contains common stuff with pipenv.
So maybe when we vendor this in pipenv we can remove it from pipenv.utils.
Also, dedup was dropped in favour of one liner.